Gems/ScriptCanvas/Code/Include/ScriptCanvas/Debugger/Debugger.h

```diff
@@ -180,7 +186,8 @@ namespace ScriptCanvas
             ActiveEntityStatusMap   m_activeEntities;
 
             AZStd::recursive_mutex m_msgMutex;
-            AzFramework::TmMsgQueue m_msgQueue;
+            AzFramework::RemoteToolsMessageQueue m_msgQueue;
+            AzFramework::IRemoteTools* m_remoteTools = nullptr;
```

Comment:
Question: is remotetool instance owned by Debugger? if not, is it better to use a sharedpointer, or use local variable on the fly instead of keeping class variable to avoid raw pointer in a stale state?

Reply:
No, IRemoteTools is an AZ::Interface whose implementation is provided by the RemoteToolsSystemComponent if it's enabled. Debugger has high volume usage of IRemoteTools so rather than do an AZ::Interface::Get for each instance, I opted to cache the pointer and just re-use it. The reference is intended to be weak so I'm just going with a raw pointer.